frontend/src/metabase/visualizations/components/settings/ChartSettingInputNumeric.tsx

```diff
@@ -1,25 +1,9 @@
-import type * as React from "react";
 import { useState } from "react";
 import _ from "underscore";
 
-import { ChartSettingNumericInput } from "./ChartSettingInputNumeric.styled";
-import type { ChartSettingWidgetProps } from "./types";
+import { NumberInput } from "metabase/ui";
 
-const ALLOWED_CHARS = [
-  "0",
-  "1",
-  "2",
-  "3",
-  "4",
-  "5",
-  "6",
-  "7",
-  "8",
-  "9",
-  ".",
-  "-",
-  "e",
```

Comment:
I think the Mantine NumberInput doesn't support scientific notation, so there's a substantive change here. See #41755

Reply:
I _think_ we should be good now

Reply:
Since we mistakenly removed scientific notation before, might you have bandwidth to throw in a unit test for it (and for allowed/disallowed chars generally)?

Reply:
there should be one there right? 

https://github.com/metabase/metabase/blob/1a5c92649a11665385b25191a0d9aeaa8330bd9e/frontend/src/metabase/visualizations/components/settings/ChartSettingInputNumeric.unit.spec.tsx#L70-L71

```
  it("allows scientific notation", async () => {
    const { input, onChange } = setup();

    await type({ input, value: "1.5e3" });
    expect(input).toHaveDisplayValue("1500");
    expect(onChange).toHaveBeenCalledWith(1.5e3);
  });
```

Reply:
It turned out that the unit tests weren't going through because they were looking for the wrong role at first. When it was `type=number`, it needed to be `spinbutton` and not `textbox`. I've reverted that change since I used `type=text` now